Hdrp StackLit Advanced Specular Occlusion Config, Iridescence control, BentNormal fixup, CoatMask, Bugfixes

slunity opened the pull request:
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
- StackLit: Added advanced specular occlusion options, fully configurable for both screen based and baked (data based) sources
- StackLit: Added iridescence control to avoid TIR due to coat, combined with artistic control of the color variation phase
- StackLit: Added coat mask
- StackLit: Added bent normal map fixup algorithm and control options
- StackLit: Fixed an important normalization issue in the stack layer variance computations (fixes an exaggerated interaction between low IOR and the effect of coatroughness)
- StackLit: Better reflection probe baking specular tint
- StackLit: Updated tests
- BSDF EvalIridescence: fixed a subtle logic vs float operation order/precision bug that could propagate NaNs. 

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?
todo

**Manual Tests**: What did you do?
- [X] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

Any test projects to go with this to help reviewers?

Launch Katana (Link on master here - update for your branch):
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=hdrp-stacklit-wrk&automation-tools_branch=master&unity_branch=trunk&sort=1-asc
---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?

**Halo Effect**: None, Low, Medium, High?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

sebastienlagarde commented:
We are waiting to land https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/3703 before processing this PR. Maybe you could already merge PR 3703 and update the code for it in the mean time?

PR 3703 still have 3 test failing (1 Hair, one VFX test and One SG lit for transmission). I am on it, so this should be green soon.

slunity commented:
> We are waiting to land #3703 before processing this PR. Maybe you could already merge PR 3703 and update the code for it in the mean time?
> PR 3703 still have 3 test failing (1 Hair, one VFX test and One SG lit for transmission). I am on it, so this should be green soon.

I'm working on it (integrating new changes for StackLit on PR 3703). I've updated to latest staging for this one (I can also open a PR for the min roughness per light for StackLit, I've updated that branch also)

sebastienlagarde commented on com.unity.render-pipelines.high-definition/Runtime/Material/StackLit/StackLit.hlsl:
```diff
@@ -1493,10 +1635,10 @@ void ComputeAdding(float _cti, float3 V, in BSDFData bsdfData, inout PreLightDat
         // Evaluate the adding operator on the normalized variance
         _s_r0m = s_ti0 + j0i*(s_t0i + s_r12 + m_rr*(s_r12+s_ri0));
-        float _s_r0i = (r0i*s_r0i + m_r0i*_s_r0m) / e_r0i;
+        float _s_r0i = (r0i*s_r0i + m_r0i*_s_r0m); // e_r0i;
```
could you comment about this change? (add a comment in the code why it is like this (in a next PR)
